morph_tool/resampling.py

```diff
@@ -29,44 +28,42 @@ def _vertex_path_lengths(points):
 
 
 def _resample_from_linear_density(points, linear_density):
-    """Creates a resampling of K-2 elements on the polyline defined by
-    N `points`, omitting the first and last values which should remain
-    unchanged. The number of elements is determined by the `linear_density`.
+    """The linear density determines the number K of the new points and point
+    properties that will be created on the polyline defined by the consecutive
+    points. The functions returns the ids and fractions, where each id i and
+    fraction t corresponds to a new point via the equation:
+
+    p' = p[i] + t * (p[i + 1] - p[i])
```

Comment:
In this equation, the left-hand side does not depend on i (nor k) while the right-hand side   does not depend on ids.

In the equation below (which I think should be totally similar), both sides depend on k and the hand-right side depends on ids.

I don't remind if it is explained somewhere yet, but the meaning of ids[k] is, if I remember correctly, the id of the starting point of the "old" segment that will contain p'[k].

Reply:
Rewrote the docstring again to make it more clear. I hope it makes more sense now.